buildSrc/subprojects/integration-testing/src/main/kotlin/org/gradle/gradlebuild/test/integrationtests/DistributionTest.kt

```diff
@@ -53,13 +55,20 @@ open class DistributionTest : Test() {
     @Internal
     val libsRepository = LibsRepositoryEnvironmentProvider(project.objects)
 
+    @get:Internal
+    @get:Option(option = "rerun", description = "Always rerun the task")
+    val rerun: Property<Boolean> = project.objects.property(Boolean::class.javaObjectType)
+
     init {
         dependsOn(Callable { if (binaryDistributions.distributionsRequired) listOf("all", "bin", "src").map { ":distributions:${it}Zip" } else null })
         dependsOn(Callable { if (binaryDistributions.binZipRequired) ":distributions:binZip" else null })
         dependsOn(Callable { if (libsRepository.required) ":toolingApi:publishLocalArchives" else null })
         jvmArgumentProviders.add(gradleInstallationForTest)
         jvmArgumentProviders.add(BinaryDistributionsEnvironmentProvider(binaryDistributions))
         jvmArgumentProviders.add(libsRepository)
+        outputs.upToDateWhen {
+            !rerun.getOrElse(false)
```

Comment:
Should we set a default instead of using `getOrElse`?

Reply:
Done.